cellarium/ml/models/contrastive_mlp.py

```diff
@@ -0,0 +1,99 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from collections.abc import Sequence
+from typing import Any, List
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from cellarium.ml.models.model import CellariumModel, PredictMixin
+from cellarium.ml.models.nt_xent import NT_Xent
+
+
+class ContrastiveMLP(CellariumModel, PredictMixin):
+    """
+    Multilayer perceptron trained with contrastive learning.
+
+    Args:
+        g_genes:
+            Number of genes in each entry (network input size).
+        hidden_size:
+            Dimensionality of the fully-connected hidden layers.
+        embed_dim:
+            Size of embedding (network output size).
+        world_size:
+            Number of devices used in training.
+        temperature:
+            Parameter governing Normalized Temperature-scaled cross-entropy (NT-Xent) loss.
+    """
+
+    def __init__(
+        self,
+        g_genes: int,
+        hidden_size: Sequence[int],
+        embed_dim: int,
+        world_size: int,
+        temperature: float = 1.0,
+    ):
+        super(ContrastiveMLP, self).__init__()
+
+        layer_list: List[nn.Module] = []
```

Comment:
Minor comment: you can create an empty `nn.Sequential()` and append to it directly:

```py
self.layers = nn.Sequential()
self.layers.append(module)
```